draft-ietf-httpbis-rfc6265bis.md

```diff
@@ -1551,76 +1555,92 @@ user agent might wish to block sending cookies during "third-party" requests
 from setting cookies (see {{third-party-cookies}}).
 
 If the user agent does attach a Cookie header field to an HTTP request, the
-user agent MUST send the cookie-string (defined below) as the value of the
-header field.
+user agent MUST send the cookie-string as the value of the
+header field. The cookie-string is computed as defined in {{retrieval-algorithm}},
+whereby the retrieval-uri is defined as the request-uri and the same-site status
+is computed for the HTTP request as defined in {{same-site-requests}}.
 
-The user agent MUST use an algorithm equivalent to the following algorithm to
-compute the cookie-string from a cookie store and a request-uri:
+### Non-HTTP APIs {#non-http}
+
+The user agent MAY implement "non-HTTP" APIs that can be used to access
+stored cookies.
 
-1.  Let cookie-list be the set of cookies from the cookie store that meets all
-    of the following requirements:
+A user agent MAY return an empty cookie-string in certain contexts, such as when
+retrieval occurs within a third-party context (see {{third-party-cookies}}).
 
-    *   Either:
+If a user agent does return cookies for a given call to a "non-HTTP" API with an
+associated Document, then the user agent MUST compute the cookie-string following
+the algorithm defined in {{retrieval-algorithm}, whereby the retrieval-uri is the
+associated Document's URL. The retrieval is same-site if the Document's "site for
+cookies" is same-site with the top-level origin as defined in {{document-requests}}.
 
-        *   The cookie's host-only-flag is true and the canonicalized
-            request-host is identical to the cookie's domain.
+### Retrieval Algorithm {#retrieval-algorithm}
```

Comment:
I think the algorithm should take an additional input of what triggered the retrieval (i.e. http request or non-http api). This piece of info is referenced in several bullet points in step 1, and it's never quite defined how to get that info from within the algorithm.

Reply:
Oh, Anne said this already. :)

Reply:
Done!